RuntimeException while resuming ViewEventActivity

From the Play Dev Console

```
java.lang.RuntimeException: Unable to resume activity {com.thebluealliance.androidclient/com.thebluealliance.androidclient.activities.ViewEventActivity}: java.lang.IllegalStateException: Cannot start this animator on a detached view!
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3103)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3134)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2481)
	at android.app.ActivityThread.-wrap11(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.IllegalStateException: Cannot start this animator on a detached view!
	at android.view.RenderNode.addAnimator(RenderNode.java:812)
	at android.view.RenderNodeAnimator.setTarget(RenderNodeAnimator.java:300)
	at android.view.RenderNodeAnimator.setTarget(RenderNodeAnimator.java:282)
	at android.animation.RevealAnimator.<init>(RevealAnimator.java:37)
	at android.view.ViewAnimationUtils.createCircularReveal(ViewAnimationUtils.java:55)
	at com.thebluealliance.androidclient.activities.MyTBASettingsActivity.closeSettingsPanel(MyTBASettingsActivity.java:320)
	at com.thebluealliance.androidclient.activities.ViewEventActivity.onNewIntent(ViewEventActivity.java:130)
	at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1212)
	at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1224)
	at android.app.ActivityThread.deliverNewIntents(ActivityThread.java:2545)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3085)
	... 10 more
```

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/MyTBASettingsActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/MyTBASettingsActivity.java
@@ -112,15 +112,10 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
 
         if (savedInstanceState != null) {
             mIsSettingsPanelOpen = savedInstanceState.getBoolean(SETTINGS_PANEL_OPEN);
-            setupFabIconForSettingsPanelOpen(mIsSettingsPanelOpen);
             if (mIsSettingsPanelOpen) {
-                mSettingsContainer.setVisibility(View.VISIBLE);
-                // Set up system UI (status bar background and icon color
-                getDrawerLayout().setStatusBarBackgroundColor(getResources().getColor(R.color
-                        .accent_dark));
-                Utilities.setLightStatusBar(getWindow(), true);
+                openSettingsPanel(false);
             } else {
-                mSettingsContainer.setVisibility(View.INVISIBLE);
+                closeSettingsPanel(false);
             }
             savedPreferenceState = savedInstanceState.getBundle(MyTBASettingsFragment.SAVED_STATE_BUNDLE);
         }
@@ -169,12 +164,12 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
             }
             if (!mIsSettingsPanelOpen) {
                 if (!mSaveInProgress) {
-                    openSettingsPanel();
+                    openSettingsPanel(true);
                 }
             } else {
                 // The user wants to save the preferences
                 // Always close the panel
-                mFabHandler.postDelayed(this::closeSettingsPanel, 1);
+                mFabHandler.postDelayed(() -> closeSettingsPanel(true), 1);
 
                 // If saving is disabled, don't attempt to save
                 if (!mIsSaveEnalbed) {
@@ -214,19 +209,42 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
      * Called when the user chooses to close the settings panel without saving their changes.
      */
     private void onSettingsCloseButtonClick() {
-        closeSettingsPanel();
+        closeSettingsPanel(true);
         // Cancel any changes made by the user
         mSettingsFragment.restoreInitialState();
     }
 
-    protected void openSettingsPanel() {
-        mSettingsFragment.restoreInitialState();
+    protected void openSettingsPanel(boolean animate) {
+        mIsSettingsPanelOpen = true;
+
+        if (mSettingsFragment != null) {
+            mSettingsFragment.restoreInitialState();
+        }
 
         // Reset the color of the button
         if (mFabColorAnimator != null) {
             mFabColorAnimator.cancel();
         }
         setFabColor(FAB_COLOR);
+
+        if (!animate) {
+            // Show the panel
+            mSettingsContainer.setVisibility(View.VISIBLE);
+
+            // Set up the status bar
+            getDrawerLayout().setStatusBarBackgroundColor(getResources().getColor(R.color
+                    .accent_dark));
+            getDrawerLayout().invalidate();
+            Utilities.setLightStatusBar(getWindow(), true);
+
+            // Dim the activity foreground
+            mForegroundDim.setAlpha(DIMMED_ALPHA);
+
+            // Configure the fab
+            setupFabIconForSettingsPanelOpen(true);
+
+            return;
+        }
 
         // Hide the button immediately
         mToggleSettingsPanelButton.setVisibility(View.GONE);
@@ -305,11 +323,30 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
                 mRunningPanelAnimation = null;
             }
         });
-
-        mIsSettingsPanelOpen = true;
     }
 
-    protected void closeSettingsPanel() {
+    protected void closeSettingsPanel(boolean animate) {
+        mIsSettingsPanelOpen = false;
+
+        if (!animate) {
+            // Hide the panel
+            mSettingsContainer.setVisibility(View.GONE);
+
+            // Set up the status bar
+            getDrawerLayout().setStatusBarBackgroundColor(getResources().getColor(R.color
+                    .primary_dark));
+            getDrawerLayout().invalidate();
+            Utilities.setLightStatusBar(getWindow(), false);
+
+            // Undim the activity foreground
+            mForegroundDim.setAlpha(UNDIMMED_ALPHA);
+
+            // Configure the fab
+            setupFabIconForSettingsPanelOpen(false);
+
+            return;
+        }
+
         int centerOfButtonOutsideX = (mToggleSettingsPanelButton.getLeft() + mToggleSettingsPanelButton.getRight()) / 2;
         int centerOfButtonOutsideY = (mToggleSettingsPanelButton.getTop() + mToggleSettingsPanelButton.getBottom()) / 2;
 
@@ -397,8 +434,6 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
                 mRunningPanelAnimation = null;
             }
         });
-
-        mIsSettingsPanelOpen = false;
     }
 
     protected void syncFabVisibilityWithMyTbaEnabled(boolean animate) {
@@ -409,11 +444,6 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
         }
     }
 
-    /**
-     * Will
-     *
-     * @param animate
-     */
     protected void showFab(boolean animate) {
         if (mFabVisible) {
             return;
@@ -574,7 +604,7 @@ public abstract class MyTBASettingsActivity extends DatafeedActivity implements 
     @Override
     public void onBackPressed() {
         if (mIsSettingsPanelOpen) {
-            closeSettingsPanel();
+            closeSettingsPanel(true);
             return;
         }
         super.onBackPressed();

--- a/android/src/main/java/com/thebluealliance/androidclient/activities/ViewEventActivity.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/activities/ViewEventActivity.java
@@ -41,6 +41,12 @@ public class ViewEventActivity extends MyTBASettingsActivity
     private FragmentComponent mComponent;
 
     /**
+     * Will be run in {@code onResume()}; used to perform UI setup that can't happen before the
+     * activity is resumed
+     */
+    private Runnable mOnNewIntentRunnable;
+
+    /**
      * Create new intent for ViewEventActivity
      *
      * @param c        context
@@ -126,12 +132,19 @@ public class ViewEventActivity extends MyTBASettingsActivity
         }
         setModelKey(mEventKey, ModelType.EVENT);
 
-        // If the settings panel was open before, close it
-        closeSettingsPanel();
+        mOnNewIntentRunnable = () -> {
+            // If the settings panel was open before, close it
+            closeSettingsPanel(false);
 
-        adapter = new ViewEventFragmentPagerAdapter(getSupportFragmentManager(), mEventKey);
-        pager.setAdapter(adapter);
-        adapter.notifyDataSetChanged();
+            // Reset the title; this will be set from the EventInfoFragment
+            setActionBarTitle("");
+
+            adapter.removeAllFragments();
+            adapter = new ViewEventFragmentPagerAdapter(getSupportFragmentManager(), mEventKey);
+            pager.setAdapter(adapter);
+            pager.setCurrentItem(mSelectedTab);
+        };
+
         Log.d(Constants.LOG_TAG, "Got new ViewEvent intent with key: " + mEventKey);
     }
 
@@ -139,6 +152,11 @@ public class ViewEventActivity extends MyTBASettingsActivity
     protected void onResume() {
         super.onResume();
         setBeamUri(String.format(NfcUris.URI_EVENT, mEventKey));
+
+        if (mOnNewIntentRunnable != null) {
+            mOnNewIntentRunnable.run();
+            mOnNewIntentRunnable = null;
+        }
     }
 
     @Override

--- a/android/src/main/java/com/thebluealliance/androidclient/adapters/ViewEventFragmentPagerAdapter.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/adapters/ViewEventFragmentPagerAdapter.java
@@ -3,6 +3,7 @@ package com.thebluealliance.androidclient.adapters;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentManager;
 import android.support.v4.app.FragmentPagerAdapter;
+import android.support.v4.app.FragmentTransaction;
 
 import com.thebluealliance.androidclient.fragments.event.EventAlliancesFragment;
 import com.thebluealliance.androidclient.fragments.event.EventAwardsFragment;
@@ -14,7 +15,14 @@ import com.thebluealliance.androidclient.fragments.event.EventStatsFragment;
 import com.thebluealliance.androidclient.fragments.event.EventTeamsFragment;
 import com.thebluealliance.androidclient.fragments.event.EventTickerFragment;
 
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
 public class ViewEventFragmentPagerAdapter extends FragmentPagerAdapter {
+
+    private List<WeakReference<Fragment>> mFragments = new ArrayList<>();
+    private FragmentManager mFragmentManager;
 
     public final String[] TITLES = {"Ticker", "Info", "Teams", "Rankings", "Matches", "Alliances", "District Points", "Stats", "Awards"};
     public static final int TAB_TICKER = 0,
@@ -31,6 +39,7 @@ public class ViewEventFragmentPagerAdapter extends FragmentPagerAdapter {
 
     public ViewEventFragmentPagerAdapter(FragmentManager fm, String eventKey) {
         super(fm);
+        mFragmentManager = fm;
         mEventKey = eventKey;
     }
 
@@ -47,27 +56,50 @@ public class ViewEventFragmentPagerAdapter extends FragmentPagerAdapter {
 
     @Override
     public Fragment getItem(int position) {
+        Fragment fragment;
         switch (position) {
             case TAB_TICKER:
-                return EventTickerFragment.newInstance(mEventKey);
+                fragment = EventTickerFragment.newInstance(mEventKey);
+                break;
             case TAB_INFO:
-                return EventInfoFragment.newInstance(mEventKey);
+                fragment = EventInfoFragment.newInstance(mEventKey);
+                break;
             case TAB_TEAMS:
-                return EventTeamsFragment.newInstance(mEventKey);
+                fragment = EventTeamsFragment.newInstance(mEventKey);
+                break;
             case TAB_RANKINGS:
-                return EventRankingsFragment.newInstance(mEventKey);
+                fragment = EventRankingsFragment.newInstance(mEventKey);
+                break;
             case TAB_MATCHES:
-                return EventMatchesFragment.newInstance(mEventKey);
+                fragment = EventMatchesFragment.newInstance(mEventKey);
+                break;
             case TAB_ALLIANCES:
-                return EventAlliancesFragment.newInstance(mEventKey);
+                fragment = EventAlliancesFragment.newInstance(mEventKey);
+                break;
             case TAB_DISTRICT_POINTS:
-                return EventDistrictPointsFragment.newInstance(mEventKey);
+                fragment = EventDistrictPointsFragment.newInstance(mEventKey);
+                break;
             case TAB_STATS:
-                return EventStatsFragment.newInstance(mEventKey);
+                fragment = EventStatsFragment.newInstance(mEventKey);
+                break;
             case TAB_AWARDS:
-                return EventAwardsFragment.newInstance(mEventKey);
+                fragment = EventAwardsFragment.newInstance(mEventKey);
+                break;
             default:
-                return new Fragment();
+                fragment = new Fragment();
         }
+        mFragments.add(new WeakReference<>(fragment));
+        return fragment;
+    }
+
+    public void removeAllFragments() {
+        FragmentTransaction transaction = mFragmentManager.beginTransaction();
+        for (WeakReference<Fragment> reference : mFragments) {
+            Fragment f = reference.get();
+            if (f != null) {
+                transaction.remove(f);
+            }
+        }
+        transaction.commit();
     }
 }

